server/src/main/java/org/cloudfoundry/identity/uaa/home/HomeController.java

```diff
@@ -136,7 +131,7 @@ public String error404(Model model) {
     @RequestMapping("/saml_error")
     public String error401(Model model, HttpServletRequest request) {
         AuthenticationException exception = (AuthenticationException) request.getSession().getAttribute(WebAttributes.AUTHENTICATION_EXCEPTION);
-        model.addAttribute("saml_error", exception.getMessage());
+        model.addAttribute("saml_error", (exception == null) ? "saml_error handler called without session stored exception" : exception.getMessage());
```

Comment:
The error text is a bit ambiguous to me. Suggest: "The server received a request for '/saml_error' but there was no SAML error information stored for the session." or some variation of that.

Also, and this is just a quibble, I personally do not like the "clause ? then : else" assignment format in most situations because I find it harder to read. You don't have to change this but I would recommend writing the same code as an if then else clause and deciding for yourself which one is easier to read.